appcues/src/main/java/com/appcues/trait/appcues/TargetInteractionTrait.kt

```diff
@@ -89,24 +98,38 @@ internal class TargetInteractionTrait(
             val clipShape = if (keyholeSettings.shape == RECTANGLE) RoundedCornerShape(keyholeSettings.cornerRadius.dp) else CircleShape
 
+            var globalOffset by remember { mutableStateOf(Offset.Zero) }
+
             Box(
                 modifier = Modifier
                     .size(width = width.dp, height = height.dp)
                     .offset(x = offsetX.dp, y = offsetY.dp)
                     .clip(clipShape)
-                    // add click listener but without any ripple effect.
-                    .then(
-                        if (actions.isNotEmpty()) {
-                            Modifier.combinedClickable(
-                                interactionSource = remember { MutableInteractionSource() },
-                                indication = null,
-                                onLongClick = actions.toLongPressMotionOrNull(actionDelegate, TARGET_LONG_PRESSED, VIEW_DESCRIPTION),
-                                onClick = actions.toTapMotionOrEmpty(actionDelegate, TARGET_TAPPED, VIEW_DESCRIPTION),
-                            )
-                        }
-                        // when no actions are listed, the default behavior is to
-                        else Modifier.pointerInput(Unit) { detectTapGestures { } }
-                    )
+                    .onGloballyPositioned { coordinates ->
+                        globalOffset = coordinates.positionInWindow()
+                    }
+                    .pointerInput(Unit) {
+                        detectTapGestures(
+                            onTap = { offset ->
+                                if (enablePassThrough) {
+                                    // If a target is allowed to pass touches through to the host application,
+                                    // we find the global offset of the tap and pass it back through to the
+                                    // View that is hosting our experience, using this tapForwardingDelegate.
+                                    // This will end up passing MotionEvents back through to the underlying views
+                                    // as if the tap was not captured at all by our Composable here.
+                                    val globalTapOffset = Offset(
+                                        x = offset.x + globalOffset.x,
+                                        y = offset.y + globalOffset.y
+                                    )
+                                    tapForwardingDelegate.onTap(globalTapOffset)
+                                }
+                                actions.toTapMotionOrEmpty(actionDelegate, TARGET_TAPPED, VIEW_DESCRIPTION).invoke()
+                            },
+                            onLongPress = {
+                                actions.toLongPressMotionOrNull(actionDelegate, TARGET_LONG_PRESSED, VIEW_DESCRIPTION)?.invoke()
```

Comment:
I included long press on iOS because it was part of the target interaction trait before, but there's not really a use case I can imagine where it'd get used. Given this is creating synthetic events, it feels a bit weird to do that with a long press, so it seems reasonable to leave it out.

Reply:
ideally we would instead fire the DOWN and the UP separately, if we would use this same approach for LONG press, imagine it takes 500ms for the USER to fire it up, it would take an additional 500ms for that event to trigger for the CUSTOMER view, right?

Reply:
yeah its basically this section of code where the synthetic `MotionEvent` is created https://github.com/appcues/appcues-android-sdk/pull/647/files#diff-eed3d6c94ac3b8e438725716643f71dd0a8e01f23818b047709860a6be149b15R152-R161

you can pass timestamp values to indicate delay between DOWN and UP there, if we wanted to handle long press.

Reply:
saving for later, we might need to use awaitPointerEventScope instead if we want to propagate other types of events https://developer.android.com/develop/ui/compose/touch-input/pointer-input/understand-gestures#event-propagation